Sources/Native/NativeMockServerWrapper.swift

```diff
@@ -2,102 +2,206 @@ import Foundation
+                    case 3:
+                        complete(.failure(.writeError("A mock server with the provided port was not found")))
+                    case 4:
+                        complete(.success("Pact verified successfully but was not written!"))
+                    default:
+                        complete(.failure(.writeError("Writing file failed, result: \(result)")))
+                    }
+                    return
+                }
+                complete(.success("Pact verified successfully!"))
+            }
+        }
+    }
+
+    private func mismatches() -> String {
+        let mismatches = mock_server_mismatches_ffi(port)
+        if let mismatches = mismatches {
+            let json = JSON(parseJSON: String(cString: mismatches))
+            var mismatches = ""
+            for (_, pathMismatch):(String, JSON) in json {
+                mismatches = "\(mismatches)\(pathMismatch["method"]) \(pathMismatch["path"]): "
+                for (_, mismatch):(String, JSON) in pathMismatch["mismatches"] {
+                    mismatches = "\(mismatches){error: \(mismatch["mismatch"]), "
+                    mismatches = "\(mismatches)expected: \(mismatch["expected"]), "
+                    mismatches = "\(mismatches)actual: \(mismatch["actual"])}"
+                }
+            }
+            return mismatches
+        } else {
+            return "Nothing received"
+        }
     }
-    return "/tmp/pacts"
-  }
-
-  func randomPort() -> Int32 {
-    return Int32(arc4random_uniform(1000) + 4000)
-  }
-
-  public func getBaseUrl() -> String {
-    return "http://localhost:\(port)"
-  }
-
-  public func setup(_ pact: Pact) -> Future<String, PactError> {
-    return Future { complete in
-      do {
-        let payload = pact.payload()
-        let jsonData = try JSONSerialization.data(withJSONObject: payload)
-        let jsonString = String(bytes: jsonData, encoding: String.Encoding.utf8)
-
-        // iOS json generation adds extra backslashes to "application/json" --> "application\\/json"
-        // causing the MockServer to fail to parse the file.
-        let sanitizedString = jsonString!.replacingOccurrences(of: "\\/", with: "/")
-        let result = create_mock_server_ffi(sanitizedString, port)
-        if result < 0 {
-          switch result {
-          case -1:
-            complete(.failure(.setupError("Mock server creation failed, pact supplied was nil")))
-          case -2:
-            complete(.failure(.setupError("Mock server creation failed, pact JSON file could not be parsed")))
-          default:
-            complete(.failure(.setupError("Mock server creation failed, result: \(result)")))
-          }
+
+    private func matched() -> Bool {
+        return mock_server_matched_ffi(port)
+    }
+
+    private func writeFile() -> Int32 {
+        guard shouldWritePacts, checkForPath() else {
+            return 4
```

Comment:
It's really hard to understand what this error would be. I'd suggest you use an enum for these.

Reply:
TODO:
agreed. same is valid for the error handling in `setup(_ pact: Pact)`. I think there was a comment on that before as well.